cmd/wallet-web/src/pages/Dashboard.vue

```diff
@@ -129,52 +127,41 @@ export default {
     ...mapGetters('agent', { getAgentInstance: 'getInstance' }),
     ...mapGetters(['getCurrentUser', 'getAgentOpts']),
     fetchAllCredentials: async function (getCredential) {
-      let { contents } = await getCredential;
+      const { contents } = await getCredential;
       console.log(`found ${Object.keys(contents).length} credentials`);
       const _filter = (id) => {
         return !contents[id].type.some((t) => filterBy.includes(t));
       };
 
-      const _createCard = (id) => {
-        return { content: contents[id], flipped: false };
-      };
-
-      this.cards = Object.keys(contents).filter(_filter).map(_createCard);
+      this.credentials = Object.keys(contents)
+        .filter(_filter)
+        .map((id) => contents[id]);
 
-      console.log(`showing ${this.cards.length} credentials`);
+      console.debug(`showing ${this.credentials.length} credentials`);
 
-      // Reading the values of the credentials and mapping it to the credential display data schemas
-      this.cjson.forEach((obj) => {
-        let flattened = {};
-        for (let credentialKey in this.cards) {
-          this.flatten(this.cards[credentialKey].content, flattened);
-          for (let credentialContent in flattened) {
-            if (
-              obj.credentialSubject.hasOwnProperty(credentialContent) &&
-              obj.schema === this.credDisplayName(this.cards[credentialKey].content)
-            ) {
-              obj.credentialSubject[credentialContent] = flattened[credentialContent];
-            }
-          }
-        }
+      this.credentials.map((credential) => {
+        const manifest = this.getManifest(credential);
+        const processedCredential = this.getCredentialDisplayData(credential, manifest);
+        this.processedCredentials.push(processedCredential);
       });
+      console.debug(`showing ${this.processedCredentials.length} credentials`);
+    },
+    getCredentialType: function (vc) {
+      return getCredentialType(vc.type);
     },
     credDisplayName: function (vc) {
       return vc.name ? vc.name : getCredentialType(vc.type);
     },
-    loadImg(imgPath) {
-      return images('./' + imgPath);
+    getCredentialDisplayData: function (vc, manifestCredential) {
+      return getCredentialDisplayData(vc, manifestCredential);
     },
-    flatten: function (json, flattened) {
-      for (let key in json) {
-        if (json.hasOwnProperty(key)) {
-          if (json[key] instanceof Object && json[key] != '') {
-            this.flatten(json[key], flattened, key);
-          } else {
-            flattened[key] = json[key];
-          }
-        }
+    getManifest: function (credential) {
+      const currentCredentialType = this.getCredentialType(credential);
+      for (const supportedCredentialType of Object.keys(credentialDisplayData)) {
```

Comment:
Why cant we use `supportedCredentialType[currentCredentialType] || supportedCredentialType.fallback`

Reply:
just curious: why we need this for-loop here to find matching manifest?

Reply:
you are right i tested without loop and made some change to || field and it works without loop